workspaces/desktop/pre-install.sh

```diff
@@ -13,7 +13,13 @@ groupadd -g 1001 prairielearner
 useradd -u 1001 -g 1001 -m -d /home/prairielearner -s /bin/bash prairielearner
 
 # install needed apps
-apt-get install xfce4-terminal firefox build-essential geany emacs-gtk vim-gtk nano gedit less -y
+apt install xfce4-terminal build-essential geany emacs-gtk vim-gtk nano gedit less -y
+
+# install firefox without snap (https://askubuntu.com/a/1369163)
+apt purge firefox -y
+apt install software-properties-common -y
+add-apt-repository -y ppa:mozillateam/ppa
+apt install firefox-esr -y
```

Comment:
What's the rationale here for installing with apt vs snap? I don't have any preference, just wondering.

Reply:
I was unable to make snap work in a container.

Reply:
Snap runs in a sandbox so it may have slower startup time for the app. The added security might cause incompatibilities with other things in rare cases (none that I know of specifically). The typical user complaint against Snap would be startup lag, the "bloat" of the install, or the fact that Canonical keeps the backend infrastructure for the snap store closed. (A more open alternative would be the Flatpak.)

I guess if the ESR version deb works, it's probably fine for the limited cases where students would need to run it in the workspace. One of the (lighter?) forks of Firefox may be even better down the road. Note that this PPA seems to be maintained by Ubuntu and Debian maintainers, but not Mozilla employees. Linux Mint also packages a deb of the latest Firefox, but I'm not sure if adding the Linux Mint repositories is a good idea.